lib/rspec/core/bisect/example_minimizer.rb

```diff
@@ -18,20 +16,65 @@ def initialize(runner, reporter)
         def find_minimal_repro
           prep
 
-          self.remaining_ids = non_failing_example_ids
+          _, duration = track_duration do
+            bisect(non_failing_example_ids)
+          end
+
+          notify(:bisect_complete, :duration => duration,
+                                   :original_non_failing_count => non_failing_example_ids.size,
+                                   :remaining_count => remaining_ids.size)
+
+          remaining_ids + failed_example_ids
+        end
+
+        def bisect(candidate_ids)
+          notify(:bisect_dependency_check_started)
+          if get_expected_failures_for?([])
+            notify(:bisect_dependency_check_finished, :dependent_failures => false)
+            self.remaining_ids = []
+            return
+          end
+          notify(:bisect_dependency_check_finished, :dependent_failures => true)
+
+          bisect_over(candidate_ids)
+        end
+
+        def bisect_over(candidate_ids)
+          return if candidate_ids.one?
 
-          each_bisect_round do |subsets|
-            ids_to_ignore = subsets.find do |ids|
+          notify(
+            :bisect_round_started,
+            :candidates_range => candidates_range(candidate_ids),
```

Comment:
IMO, this would read more naturally as `candidate_range` rather than `candidates_range` -- kinda like you hear people say "age range" but not "ages range".